moodler/moodle.py

```diff
@@ -67,23 +82,22 @@ def submissions_statistics(course_id, is_verbose=False, download_folder=None):
     total_submissions = 0
     total_ungraded = 0
     total_resubmissions = 0
+    total_unreleased = 0
     assignments_statistics = {}
 
     assignments = get_assignments(course_id)
     users_map = get_students(course_id)
 
     for assignment in assignments:
-        current_assignment_submissions_amount = len(assignment.submissions)
-        current_assignment_ungraded = assignment.ungraded()
-        current_assignment_ungraded_amount = len(current_assignment_ungraded)
-
-        current_assignment_resubmissions_amount = 0
+        submissions_amount = len(assignment.submissions)
+        submitted_submissions = assignment.submitted()
+        ungraded_submissions = assignment.ungraded()
+        ungraded_amount = len(ungraded_submissions)
+        resubmissions_amount = len([s for s in submitted_submissions if s.resubmitted])
+        unreleased_amount = len([s for s in submitted_submissions if s.released])
         ungraded_ignored = []
 
-        for submission in current_assignment_ungraded:
-            if submission.resubmitted:
-                current_assignment_resubmissions_amount += 1
```

Comment:
Previously, the logic counted the number of resubmissions for the ungraded submissions only.

Reply:
Counting the number of resubmissions for the ungraded submissions only is what we need here. We use this to prioritize which exercises to check first. So if we see that all submissions are just resubmissions, we might prioritize new submissions in other exercises first.